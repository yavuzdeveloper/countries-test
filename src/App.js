import React, { useEffect, useState } from "react";
import "./App.css";
import axios from "axios";
import Country from "./Country"


export default function App() {
  const [countries, setCountries] = useState([]);
  const [loaded, setLoaded] = useState(false);

  useEffect(() => {
    axios
      .get("https://restcountries.eu/rest/v2/all")
      .then(response => {
        setCountries(response.data)
         setLoaded(true)
      })
      .catch(error => 
        console.log({ error })
      );
  }, []);



  return (
    <div className="App">
      <h1 className="title"> 
        <img 
          src="https://avatars3.githubusercontent.com/u/60869810?v=4" 
          alt="react-logo" 
          style={{ width: "20px", padding:"0 10px 0 0" }}
        />
        Countries
      </h1>
      { loaded ? ( <div> 
      { countries && countries.map(country => ( 
        <div>
          <Country  country={country}/> 
        </div>
        )
      )}
    </div>
    ) : ( <p>loading...
          {/* <img 
            src= "https://media.giphy.com/media/y1ZBcOGOOtlpC/source.gif" 
            alt="imgLoading"       
          /> */}
        </p>)
      }
    </div>
  );
}
