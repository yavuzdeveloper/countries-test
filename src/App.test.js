import { render,cleanup, waitFor, screen } from '@testing-library/react';
import App from './App';
import mockedAxios from 'axios';

import axios from 'axios';
import TestRenderer from "react-test-renderer";



test('renders react-logo', () => {
  render(<App />);
  const logoReact = screen.getByRole("img");
  
  expect(logoReact).toHaveAttribute("src", "https://avatars3.githubusercontent.com/u/60869810?v=4")
  expect(logoReact).toHaveAttribute('alt', "react-logo");
});


test('renders Countires title', () => {
  render(<App />);
  const titleElement = screen.getByText(/countries/i);

  expect(titleElement).toBeInTheDocument();
  expect(titleElement).toHaveTextContent(/countries/i);
});


test('renders loading...', () => {
  render(<App />);
  const loadingElement = screen.getByText(/loading/i);
  expect(loadingElement).toBeInTheDocument();
  expect(loadingElement).toHaveTextContent(/loading.../i)
});



//axios-mock;

// afterEach(cleanup);

// jest.mock('axios');// added??

// test('mocking axios request', async () => {
//   const data = {
//     countries: [
//       { name: 'country1' },
//       { name: 'country2' },
//       { name: 'country3' }
//     ]
//   };

//   mockedAxios.get.mockResolvedValueOnce(data);
//   const { getByText } = render(<App />);

//   await waitFor(() => {
//     expect(getByText('country1')).toBeInTheDocument();
//   });
// });

//***************
jest.mock('axios');

test("fetch", () => {
  const response = {
    countries: [
      { name: 'country1' },
      { name: 'country2' },
      { name: 'country3' }
    ]
  }
  axios.get.mockResolvedValue(response);

    const component = TestRenderer.create(
      <App />
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})
