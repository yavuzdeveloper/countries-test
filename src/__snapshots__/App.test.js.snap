// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetch 1`] = `
<div
  className="App"
>
  <h1
    className="title"
  >
    <img
      alt="react-logo"
      src="https://avatars3.githubusercontent.com/u/60869810?v=4"
      style={
        Object {
          "padding": "0 10px 0 0",
          "width": "20px",
        }
      }
    />
    Countries
  </h1>
  <p>
    loading...
  </p>
</div>
`;
